src/coreclr/vm/multicorejit.cpp

```diff
@@ -1149,12 +1139,9 @@ void MulticoreJitManager::SetProfileRoot(const WCHAR * pProfilePath)
 
 #endif
 
-    if (g_SystemInfo.dwNumberOfProcessors >= 2)
```

Comment:
Agreed @noahfalk.  @gbalykov, can you share more details about this change?  My expectation is that if there is only one processor, then playing back a Multi-Core JIT profile is likely to impact start-up negatively.

Reply:
On arm devices cpus might be enabled dynamically (cpu hotplug). Thus, during coreclr_initialize there might be only one enabled cpu, yet there'll be more when there's enough work to do

Reply:
Thanks @gbalykov.  My first thought is that if we're going to remove this check, we're going to want to limit the removal to Linux arm builds.  It looks like per https://community.arm.com/developer/tools-software/oss-platforms/w/docs/529/cpuidle-hotplug, hotplug is only available on Linux.

My next thought is that I'm wanting to understand more about the hotplug behavior.  Multi-Core JIT looks at the CPU count once when it gets invoked, and that's it.  It sounds like we might need to modify Multi-Core JIT to be hotplug aware (if that's possible) so that we don't needlessly do aggressive assembly loading and jitting on the critical path if we want to support the hotplug scenario on arm.  Is it possible to be hotplug aware?

Reply:
Another option may be to make it configurable for now via `clrconfigvalues.h` and keep the default as it is currently. On CPU-limited environments where there is actually only one processor available, the background thread could otherwise steal a fair bit of CPU time from foreground work.

Reply:
> Another option may be to make it configurable for now via clrconfigvalues.h

I've added this solution instead of check removal. By default 2 cpus will still be needed